core/offchain/impl/offchain_worker_impl.cpp

```diff
@@ -22,74 +22,55 @@
 
 namespace kagome::offchain {
 
+  size_t OffchainWorkerImpl::ocw_counter_ = 0;
+
   OffchainWorkerImpl::OffchainWorkerImpl(
       const application::AppConfiguration &app_config,
       std::shared_ptr<clock::SystemClock> clock,
-      std::shared_ptr<crypto::Hasher> hasher,
       std::shared_ptr<storage::SpacedStorage> storage,
       std::shared_ptr<crypto::CSPRNG> random_generator,
       std::shared_ptr<api::AuthorApi> author_api,
       const network::OwnPeerInfo &current_peer_info,
       std::shared_ptr<OffchainPersistentStorage> persistent_storage,
-      std::shared_ptr<runtime::Executor> executor,
-      const primitives::BlockHeader &header,
       std::shared_ptr<OffchainWorkerPool> ocw_pool)
       : app_config_(app_config),
         clock_(std::move(clock)),
-        hasher_(std::move(hasher)),
         random_generator_(std::move(random_generator)),
         author_api_(std::move(author_api)),
         current_peer_info_(current_peer_info),
         persistent_storage_(std::move(persistent_storage)),
-        executor_(std::move(executor)),
-        header_(header),
         ocw_pool_(std::move(ocw_pool)),
         log_(log::createLogger(
-            "OffchainWorker#" + std::to_string(header_.number), "offchain")) {
+            "OffchainWorker#" + std::to_string(++ocw_counter_), "offchain")) {
```

Comment:
Ambigous label, looks like previously used block number

Reply:
Yes and not. It's just counter ocw, not related to block. Now is possible to run several or no one ocw based on block context.